src/swapRouter.ts

```diff
@@ -172,28 +193,40 @@ export abstract class SwapRouter extends SelfPermit {
 
         if (outputIsEther) {
           calldatas.push(
-            SwapRouter.INTERFACE.encodeFunctionData('unwrapWETH9WithFee', [amountOut, recipient, fee, feeRecipient])
+            SwapRouter.INTERFACE.encodeFunctionData('unwrapWETH9WithFee', [
+              toHex(totalAmountOut.quotient),
+              recipient,
+              fee,
+              feeRecipient
+            ])
           )
         } else {
           calldatas.push(
             SwapRouter.INTERFACE.encodeFunctionData('sweepTokenWithFee', [
-              trade.route.tokenPath[trade.route.tokenPath.length - 1].address,
-              amountOut,
+              sampleTrade.route.tokenPath[sampleTrade.route.tokenPath.length - 1].address,
+              toHex(totalAmountOut.quotient),
               recipient,
               fee,
               feeRecipient
             ])
           )
         }
       } else {
-        calldatas.push(SwapRouter.INTERFACE.encodeFunctionData('unwrapWETH9', [amountOut, recipient]))
+        calldatas.push(
+          SwapRouter.INTERFACE.encodeFunctionData('unwrapWETH9', [toHex(totalAmountOut.quotient), recipient])
+        )
       }
     }
 
+    // refund
```

Comment:
shouldn't this reorder have changed the output for existing tests?

Reply:
It looks like the combination that would be reordered is "ETH in, exact output, with fee" which is a combination we didn't have in the original tests. Added tests

Reply:
ah understood, thanks